server/src/main/scala/actors/FlightsStateActor.scala

```diff
@@ -36,22 +36,27 @@ object FlightsStateActor {
                         now: () => SDateLike,
                         queues: Map[Terminal, Seq[Queue]],
                         expireAfterMillis: Int,
-                        legacyDataCutoff: SDateLike): Props = {
+                        legacyDataCutoff: SDateLike,
+                        replayMaxCrunchStateMessages: Int): Props = {
     if (isNonLegacyRequest(pointInTime, legacyDataCutoff))
-      Props(new FlightsStateReadActor(now, expireAfterMillis, pointInTime.millisSinceEpoch, queues, legacyDataCutoff))
+      Props(new FlightsStateReadActor(now, expireAfterMillis, pointInTime.millisSinceEpoch, queues, legacyDataCutoff, replayMaxCrunchStateMessages))
     else
-      Props(new CrunchStateReadActor(pointInTime, expireAfterMillis, queues, message.from, message.to))
+      Props(new CrunchStateReadActor(pointInTime, expireAfterMillis, queues, message.from, message.to, replayMaxCrunchStateMessages))
   }
 }
 
-class FlightsStateActor(val now: () => SDateLike, expireAfterMillis: Int, queues: Map[Terminal, Seq[Queue]], legacyDataCutoff: SDateLike)
+class FlightsStateActor(val now: () => SDateLike,
+                        expireAfterMillis: Int,
+                        queues: Map[Terminal, Seq[Queue]],
+                        legacyDataCutoff: SDateLike,
+                        replayMaxCrunchStateMessages: Int)
   extends PersistentActor with RecoveryActorLike with PersistentDrtActor[FlightsWithSplits] {
 
   import FlightsStateActor._
 
   override def persistenceId: String = "flights-state-actor"
 
-  override val maybeSnapshotInterval: Option[Int] = Option(5000)
+  override val maybeSnapshotInterval: Option[Int] = Option(1000)
```

Comment:
I'm reducing this to 1000 from 5000 as 5000 seems a little on the high side which could slow down historic data recovery